js/utils.js

```diff
@@ -792,7 +792,10 @@ Fliplet.Registry.set('dynamicListUtils', (function () {
             return;
           }
 
-          if (entryData[column] && file.name.indexOf(entryData[column]) !== -1) {
+          var lastDotInNameIndex = file.name.lastIndexOf('.');
+          var nameWithoutExt = file.name.substring(0, lastDotInNameIndex);
```

Comment:
What happens when `lastDotInNameIndex` is `-1`?

![image](https://user-images.githubusercontent.com/290733/72515648-2a7f2d80-3848-11ea-8374-ffb52e6d270e.png)

This looks like it won't work when the file name doesn't contain a `.`.

We should also be able to extract this with a regexp.

Reply:
@tonytlwu In line 798 we first compare the file name without extension and then with removed extension. I tested this and it works fine.

Reply:
I see. Then this is confusing because when a `.` isn't found, the line `nameWithoutExt = file.name.substring(0, lastDotInNameIndex)` not only doesn't make sense, but is also unnecessary.

There should be regex that's available to get the file name, e.g.

```js
var fileName = file.name.match(/(.+?)(?:\.[^\.]*$|$)/)[1];
```

If this doesn't work, research and find the appropriate solution. Otherwise, this logic feels odd as I described above.

Reply:
updated to use regexp